libasynql/src/poggit/libasynql/base/DataConnectorImpl.php

```diff
@@ -200,11 +201,11 @@ public function executeSelectRaw(string $query, array $args = [], ?callable $onS
 	public function asyncSelect(string $queryName, array $args = []) : Generator{
 		$onSuccess = yield Await::RESOLVE;
 		$onError = yield Await::REJECT;
-		$this->executeSelect($queryName, $args, static function(array $rows, SqlColumnInfo $columns) use($onSuccess) : void{
+		$this->executeSelect($queryName, $args, static function(array $rows) use($onSuccess) : void{
```

Comment:
I removed $columns parameter because if you want to use it, you should use `asyncSelectWithInfo()`

Reply:
ok, so what would be the purpose of passing two parameters when only one will be used? This forces the user to needlessly pass a second parameter in my opinion

Reply:
This does not *force* the user to pass a second parameter, because we are the user here. The definition of `executeSelect` stated clearly that two parameters will be passed, and only accepting one parameter is an antipattern in my opinion because that is not the same function signature.

Reply:
Also, this is not "passing" two parameters. This is "accepting" parameters.

By the way, in await-generator 4.0.0, you cannot pass `yield Await::RESOLVE` to callers that pass more than one parameter, because that most likely implies a bug where you forgot to handle the remaining parameters.

Reply:
Relying on the fact that passing two parameters automatically gets truncated into one parameter would cause forward compatibility problems. For example, if the caller changes to only passing one parameter (a different one), this is a BC break that should have warranted a crash, but since you always only accepted one parameter, you silently accepted the BC break without getting a crash, and hence lead to further incorrect behaviour that might damage your logic. This is dangerous.

Reply:
I don't force users to always accept two arguments, but my code style requires that closures take as many parameters as they are required to, instead of silently truncating.